# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1259218418 -3600
#      Thu Nov 26 07:53:38 2009 +0100
# Node ID b5aa5d9ee6b96bc411feb753b66b57d36379a420
# Parent  763906db555e55d4d827fb4385b1bb1db515f3e7
make ismatrix consistent for empty matrices

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-11-26  Jaroslav Hajek  <highegg@gmail.com>
+
+	* data.cc (Fismatrix): Return true for empty matrices as well.
+
 2009-11-25  Jaroslav Hajek  <highegg@gmail.com>
 
 	* DLD-FUNCTIONS/chol.cc (get_chol_l): New helper function.
 	(Fchol): Use it to set MatrixType for lower triangular factors as
 	well. Use default octave_idx_type->octave_value conversion.
 
 2009-11-24  Jaroslav Hajek  <highegg@gmail.com>
 
diff --git a/src/data.cc b/src/data.cc
--- a/src/data.cc
+++ b/src/data.cc
@@ -3351,43 +3351,37 @@ Return 1 if @var{a} is a matrix.  Otherw
 @end deftypefn")
 {
   octave_value retval = false;
 
   if (args.length () == 1)
     {
       octave_value arg = args(0);
 
-      if (arg.is_scalar_type () || arg.is_range ())
-	retval = true;
-      else if (arg.is_matrix_type ())
-	retval = (arg.rows () >= 1 && arg.columns () >= 1);
+      retval = arg.is_matrix_type () || arg.is_scalar_type () || arg.is_range ();
     }
   else
     print_usage ();
 
   return retval;
 }
 
 /*
 
+%!assert(ismatrix ([]));
 %!assert(ismatrix (1));
 %!assert(ismatrix ([1, 2, 3]));
 %!assert(ismatrix ([1, 2; 3, 4]));
-
-%% Yes, this is right, ismatrix() checks for non-empty matrices.
-%!assert(ismatrix ([]), false);
-
+%!assert(ismatrix (zeros (3, 2, 4)));
+
+%!assert(ismatrix (single([])));
 %!assert(ismatrix (single(1)));
 %!assert(ismatrix (single([1, 2, 3])));
 %!assert(ismatrix (single([1, 2; 3, 4])));
 
-%% Yes, this is right, ismatrix() checks for non-empty matrices.
-%!assert(ismatrix (single([])), false);
-
 %!assert(ismatrix ("t"), false);
 %!assert(ismatrix ("test"), false);
 %!assert(ismatrix (["test"; "ing"]), false);
 
 %!test
 %! s.a = 1;
 %! assert(ismatrix (s), false);
 
